### PR TITLE
MGDSTRM-9180-Unmapped URLs result in unhandled HTTP status 500 

### PR DESCRIPTION
**Pull request by biswassri**
This is to fix admin server error handling for `NotFoundException` to a new generic error code that returns a `404` status.

### CODE REVIEW

File: kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import org.apache.kafka.common.errors.InvalidTopicException;
 import org.apache.kafka.common.errors.LeaderNotAvailableException;
 import org.apache.kafka.common.errors.PolicyViolationException;
+import org.apache.kafka.common.errors.ResourceNotFoundException;
```
Comment (MikeEdgar) [inline kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java:15]: @biswassri , the exception to look for is `javax.ws.rs.NotFoundException`. A test like the following in `RestEndpointTestIT` demonstrates the issue (will fail on `main` branch).

```java
    @Test
    void testUnknownEndpointReturnsNotFound() {
        given()
            .log().ifValidationFails()
            .when()
                .get("/foo/bar")
            .then()
                .log().ifValidationFails()
                .statusCode(Status.NOT_FOUND.getStatusCode());
    }
```

Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java:15]: Oh this is the test that used to get skipped on the main. Right! I'll try this

Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java]: @MikeEdgar  I updated the class and tried this test. It passes with these PR changes. Do you want me to commit this `testUnknownEndpointReturnsNotFound()` to the PR as well?

Comment (MikeEdgar) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java]: > Do you want me to commit this `testUnknownEndpointReturnsNotFound()` to the PR as well?

Yes, please. Sorry, I thought I had answered you on this.

Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java]: No worries Mike! I'll go ahead and add this in.

File: kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java [hunk 2]
```diff
@@ -76,6 +77,7 @@ public ErrorHandler(Class<T> causeType, Function<T, ResponseBuilder> mapper) {
             // 404 Not Found
             entry(UnknownTopicOrPartitionException.class, thrown -> errorResponse(thrown, ErrorType.TOPIC_NOT_FOUND)),
             entry(GroupIdNotFoundException.class, thrown -> errorResponse(thrown, ErrorType.GROUP_NOT_FOUND)),
+            entry(NotFoundException.class, thrown -> errorResponse(thrown, ErrorType.ERROR_NOT_FOUND)),
```
Comment (MikeEdgar) [inline kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java:80]: Change to use the new error type
```suggestion
            entry(NotFoundException.class, thrown -> errorResponse(thrown, ErrorType.RESOURCE_NOT_FOUND)),
```

File: systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RestEndpointTestIT.java [hunk 3]
```diff
@@ -697,4 +697,14 @@ void testOptionsRequestIncludesTwoHourMaxAge() {
...
\ No newline at end of file
+    @Test
+    void testUnknownEndpointReturnsNotFound() {
+        given()
+            .log().ifValidationFails()
+            .when()
+                .get("/foo/bar")
+            .then()
+                .log().ifValidationFails()
+                .statusCode(Status.NOT_FOUND.getStatusCode());
```
Comment (MikeEdgar) [inline systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RestEndpointTestIT.java:708]: Better to test not just the status but also that the body matches what is expected for the new `ErrorType`.
```suggestion
        final ErrorType expectedError = ErrorType.RESOURCE_NOT_FOUND;

        given()
            .log().ifValidationFails()
            .when()
                .get("/foo/bar")
            .then()
                .log().ifValidationFails()
            .assertThat()
                .statusCode(expectedError.getHttpStatus().getStatusCode())
                .body("", matchesError(expectedError));
```

**Review by MikeEdgar — CHANGES_REQUESTED** [review]
One bug to fix and a suggestion to enhance the test.

File: kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java (hunk 2 above)
Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/handlers/CommonHandler.java]: Oh sorry, my mistake. Thanks for catching this Mike.

**Review by MikeEdgar — APPROVED** [review]
Looks good, thanks @biswassri

**Review by rareddy — APPROVED** [review]
lgtm

**Pull request merged**